src/snowflake/cli/api/sql_execution.py

```diff
@@ -128,56 +128,21 @@ def check_database_and_schema_provided(self, name: Optional[str] = None) -> None
         """
         Checks if a database and schema are provided, either through the connection context or a qualified name.
         """
-        if name:
-            _, schema, database = from_qualified_name(name)
-        else:
-            schema, database = None, None
-        schema = schema or self._conn.schema
-        database = database or self._conn.database
-        if not database:
+        fqn = FQN.from_string(name).using_connection(self._conn)
+        if not fqn.database:
             raise DatabaseNotProvidedError()
-        if not schema:
+        if not fqn.schema:
             raise SchemaNotProvidedError()
 
-    def to_fully_qualified_name(
-        self, name: str, database: Optional[str] = None, schema: Optional[str] = None
-    ):
-        current_parts = name.split(".")
-        if len(current_parts) == 3:
-            # already fully qualified name
-            return name.upper()
-
-        if not database:
-            if not self._conn.database:
-                raise DatabaseNotProvidedError()
-            database = self._conn.database
-
-        if len(current_parts) == 2:
-            # we assume name is in form of `schema.object`
-            return f"{database}.{name}".upper()
-
-        schema = schema or self._conn.schema or "public"
-        database = database or self._conn.database
-        return f"{database}.{schema}.{name}".upper()
-
-    @staticmethod
-    def get_name_from_fully_qualified_name(name):
-        """
-        Returns name of the object from the fully-qualified name.
-        Assumes that [name] is in format [[database.]schema.]name
-        """
-        return from_qualified_name(name)[0]
-
     @staticmethod
-    def _qualified_name_to_in_clause(name: str) -> Tuple[str, Optional[str]]:
-        unqualified_name, schema, database = from_qualified_name(name)
-        if database:
-            in_clause = f"in schema {database}.{schema}"
-        elif schema:
-            in_clause = f"in schema {schema}"
+    def _qualified_name_to_in_clause(identifier: FQN) -> Tuple[str, Optional[str]]:
+        if identifier.database:
```

Comment:
```suggestion
        if identifier.database and identifier.schema:
```

Maybe I don't see sth but I think it's possible to create FQN with db but without schema.

Reply:
I mean it's possible to create FQN class instance without schema.

Reply:
Fixed